Added dialogue choice delay

laikasghost opened the pull request:
Added IEnumerator to DialogueManager to create 1 second delay between reaching decision tree and being able to click on things

Commit:
Added dialogue choice delay

Added IEnumerator to DialogueManager to create 1 second delay between reaching decision tree and being able to click on things

SiennaC reviewed (changes requested):
This doesn't quite do what we need it to. The desired functionality is as follows:
Display dialogue lines and dialogue choices normally (note that empty dialogue lines are skipped.)
Do not allow the player to click to advance dialogue (either lines or choices) for a brief period after the dialogue is shown.

I recommend using UIManager.BlockAllInput to disallow clicking through dialogue. You shouldn't need to change/move the code that's already there, probably just add an enum to block input, then unblock it after a brief wait. Make sure this happens for both choices and lines.

Note also that the amount of time dialogue advancement is blocked should be easily customizable. (This is something we're definitely going to want to tweak to make sure it feels ok.) I recommend a const in DialogueManager for this.

Commit:
Merge branch 'master' into CaitiBranchII

Commit:
Edited dialogue delay

Added const float clickDelay, changed implementation of IEnum, added delay to dialogue as well as choices

SiennaC commented on Assets/Scripts/Dialogue/DialogueManager.cs:
```diff
@@ -10,6 +10,8 @@
 public class DialogueManager : Manager<DialogueManager> {
     private const string SPEECH_BUBBLE_TAG = "Speech bubble";
 
+	public const float clickDelay = 0.3f;
```
Let's make this private, since we shouldn't need to access it elsewhere, and match case with our other constants (i.e. CLICK_DELAY).

SiennaC commented on Assets/Scripts/Dialogue/DialogueManager.cs:
```diff
@@ -259,25 +261,36 @@ public class DialogueManager : Manager<DialogueManager> {
             gameObject.SetActive(false);
             DialogueManager.Next(line);
         }
+		StartCoroutine (DelayClick ());
```
The if/else statement preceding this is designed to skip dialogue lines that have no text. If a line is skipped, it should not trigger the click delay. (i.e. this line should be inside of the if statement).

Commit:
Made requested changes

Changed public const to private, changed clickDelay to CLICK_DELAY, moved coroutine
